datafusion/common/src/dfschema.rs

```diff
@@ -677,7 +687,7 @@ mod tests {
         // lookup with unqualified name "t1.c0"
         let err = schema.index_of_column(&col).err().unwrap();
         assert_eq!(
-            "Schema error: No field named 't1.c0'. Valid fields are 't1'.'c0', 't1'.'c1'.",
+            r#"Schema error: No field named "t1.c0". Valid fields are "t1"."c0", "t1"."c1"."#,
```

Comment:
this was a separate change i decided to add, mainly since i believe having quoted identifiers allows for more robust messages as itll properly account for any special characters within and allows users to clearly delineate identifiers in that case

e.g. in old way a user could have an ident like `t1'.'t2` which would show in error as `'t1'.'t2'` even though it is a single ident, it looks like two

Reply:
Yeah this did come to mind, but:

- Either have to store alongside the name that it was normalized
- Or each time quoting string have to do a pass over it to check for special characters/uppercase characters

1st one I don't really like as have to add more to TableReference (and related structs)

2nd one would be simpler to implement, maybe I'm overthinking the impact of the overhead 😅

Reply:
raised ticket: https://github.com/apache/arrow-datafusion/issues/5523